packages/flutter_tools/lib/src/windows/build_windows.dart

```diff
@@ -147,6 +163,8 @@ Future<void> _runCmakeGeneration({
         buildDir.path,
         '-G',
         generator,
+        '-A',
+        _getCmakeArch(targetPlatform: targetPlatform),
```

Comment:
Please update the test as well: https://github.com/flutter/flutter/blob/35afe1bdac653c752c5794cf877b64238a4c9e4c/packages/flutter_tools/test/commands.shard/hermetic/build_windows_test.dart#L74